packages/uploadthing/src/internal/multi-part.ts

```diff
@@ -113,7 +113,7 @@ export async function uploadPartWithProgress(
         // Add a delay before retrying (exponential backoff can be used)
         const delay = Math.pow(2, retryCount) * 1000;
         await new Promise((res) => setTimeout(res, delay));
-        await uploadPartWithProgress(opts, retryCount + 1); // Retry the request
+        resolve(await uploadPartWithProgress(opts, retryCount + 1)); // Retry the request
```

Comment:
how funny that when validating the new solution i think i found a bug in the current solution xddd

the promise never resolved if it retried so it just hang forever 😅

Reply:
😅